tools/mmp/aot.cs

```diff
@@ -0,0 +1,180 @@
+		}
+	}
+
+	public delegate int RunCommandDelegate (string path, string args, string[] env = null, StringBuilder output = null, bool suppressPrintOnErrors = false);
+
+	public class AOTCompiler
+	{
+		enum AotType {
+			Default,
+			None,
+			All,
+			SDK,
+			Explicit
+		}
+
+		AotType aotType = AotType.Default;
+		public bool IsAOT => aotType != AotType.Default && aotType != AotType.None; 
+
+		// Set to Key -> True when we've seen a given include/exclude during compile to catch errors
+		Dictionary <string, bool> includedAssemblies = new Dictionary <string, bool> ();
+		Dictionary <string, bool> excludedAssemblies = new Dictionary <string, bool> ();
+
+		// Allows tests to stub out actual compilation and parallelism
+		public RunCommandDelegate RunCommand { get; set; } = Driver.RunCommand; 
+		public ParallelOptions ParallelOptions { get; set; } = new ParallelOptions () { MaxDegreeOfParallelism = Driver.Concurrency };
+
+		public string Quote (string f) => Driver.Quote (f);
+
+		public void Parse (string options)
+		{
+			// Syntax - all,sdk,none or "" if explicit then optional list of +/-'ed assemblies
+			// Sections seperated by ,
+			foreach (var option in options.Split (',')) {
+				switch (option) {
+				case "none":
+					aotType = AotType.None;
+					continue;
+				case "all":
+					aotType = AotType.All;
+					continue;
+				case "sdk":
+					aotType = AotType.SDK;
+					continue;
+				}
+
+				if (option.StartsWith ("+", StringComparison.Ordinal)) {
+					if (aotType == AotType.Default)
+						aotType = AotType.Explicit;
+					includedAssemblies.Add (option.Substring (1), false);
+					continue;
+				}
+				if (option.StartsWith ("-", StringComparison.Ordinal)) {
+					if (aotType == AotType.Default)
+						aotType = AotType.Explicit;
+					excludedAssemblies.Add (option.Substring (1), false);
+					continue;
+				}
+				throw new MonoMacException (20, true, "The valid options for '{0}' are '{1}'.", "--aot", "none, all, sdk, and an explicit list of assemblies.");
+			}
+			if (aotType == AotType.Default)
+				throw new MonoMacException (20, true, "The valid options for '{0}' are '{1}'.", "--aot", "none, all, sdk, and an explicit list of assemblies.");
+		}
+
+		List<string> GetFilesToAOT (IFileEnumerator files)
+		{
+			var aotFiles = new List<string> ();
+			foreach (var file in files.Files) {
+				string fileName = Path.GetFileName (file);
+				string extension = Path.GetExtension (file);
+				if (extension != ".exe" && extension != ".dll")
+					continue;
+
+				if (excludedAssemblies.ContainsKey (fileName)) {
+					excludedAssemblies[fileName] = true;
+					continue;
+				}
+
+				if (includedAssemblies.ContainsKey (fileName)) {
+					includedAssemblies[fileName] = true;
+					aotFiles.Add (file);
+					continue;
+				}
+
+				switch (aotType) {
+				case AotType.All:
+					aotFiles.Add (file);
+					break;
+				case AotType.SDK:
+					if (fileName == "Xamarin.Mac.dll" || fileName == "System.dll" || fileName == "mscorlib.dll")
```

Comment:
Let's not overload (and confuse) the meaning of **SDK** in the tooling. Rename it to something else, e.g. `core` ?

FWIW `sdk` might be an valid (we lack data to see if it's a good one) option, it's definitively a good one for the linker (and we should avoid confusion between them).